src/Microsoft.DotNet.TestFramework/TestAssetInfo.cs

```diff
@@ -57,6 +80,8 @@ internal TestAssetInfo(DirectoryInfo root, string assetName)
                 _dataDirectory
             };
 
+            RemoveDataDirectoryIfAssetChanged();
```

Comment:
I agree with the functionality. This was just a comment on code structure. Constructors shouldn't have side effects. I'd prefer to avoid making any changes to the file system until some method is invoked.